sled-agent/src/instance.rs

```diff
@@ -792,17 +796,36 @@ impl Instance {
         target: InstanceRuntimeStateRequested,
     ) -> Result<InstanceRuntimeState, Error> {
         let mut inner = self.inner.lock().await;
-        if let Some(action) = inner
-            .state
-            .request_transition(&target)
-            .map_err(|e| Error::Transition(e))?
-        {
-            info!(
-                &inner.log,
-                "transition to {:?}; action: {:#?}", target, action
-            );
-            inner.take_action(action).await?;
+        match target.run_state {
```

Comment:
it'd be nice to have just a single `propolis_state_put` command and just match to get the propolis `InstanceStateRequested` variant. Helps a bit with rightward drift.

That would possibly change the order of calls to `propolis_state_put` and `transition` for some arms I guess. But we do have the `inner` lock here so it's not like any observed propolis state can prempt us.

Reply:
Cleaned this up a bit in commit 4e7373c2. I think it's useful to keep the Propolis request and the internal state transition in order so that we don't have to undo state transitions when our Propolis calls fail (it's safe to do that from a synchronization point of view but still adds a bit of friction).

FWIW, this path will also get reworked somewhat in a [subsequent commit](https://github.com/oxidecomputer/omicron/commit/504fa56f1c33e62ba2786f1f7386c9fdffd8a5dd#diff-e58fdced2e69108a490916cca62b391c44fdde81d564de6f86b8d4e1febc5998R597).